### ISSUE TITLE
Route::resource with 'as' parameter should set whole route to the value of 'as'

### ISSUE THREAD

**Issue by w4** [laravel/framework#12068]
Currently:

    Route::group(['prefix' => 'admin', 'as' => 'admin::'], function () {
        Route::resource('user', 'UserController');
    });

Will become:
```
|        | GET|HEAD                       | admin/user                                                  | admin::admin.user.index            | App\Http\Controllers\Admin\UsersController@index
|        | POST                           | admin/user                                                  | admin::admin.user.store            | App\Http\Controllers\Admin\UsersController@store
|        | GET|HEAD                       | admin/user/create                                           | admin::admin.user.create           | App\Http\Controllers\Admin\UsersController@create
|        | DELETE                         | admin/user/{user}                                           | admin::admin.user.destroy          | App\Http\Controllers\Admin\UsersController@destroy
|        | GET|HEAD                       | admin/user/{user}                                           | admin::admin.user.show             | App\Http\Controllers\Admin\UsersController@show
|        | PUT|PATCH                      | admin/user/{user}                                           | admin::admin.user.update           | App\Http\Controllers\Admin\UsersController@update
|        | GET|HEAD                       | admin/user/{user}/edit                                      | admin::admin.user.edit             | App\Http\Controllers\Admin\UsersController@edit
```

and

    Route::group(['prefix' => 'admin', 'as' => 'admin::'], function () {
        Route::resource('user', 'UserController', ['as' => 'user']);
    });

will become

```
|        | GET|HEAD                       | admin/user                                                  | admin::user.admin.user.index            | App\Http\Controllers\Admin\UsersController@index
|        | POST                           | admin/user                                                  | admin::user.admin.user.store            | App\Http\Controllers\Admin\UsersController@store
|        | GET|HEAD                       | admin/user/create                                           | admin::user.admin.user.create           | App\Http\Controllers\Admin\UsersController@create
|        | DELETE                         | admin/user/{user}                                           | admin::user.admin.user.destroy          | App\Http\Controllers\Admin\UsersController@destroy
|        | GET|HEAD                       | admin/user/{user}                                           | admin::user.admin.user.show             | App\Http\Controllers\Admin\UsersController@show
|        | PUT|PATCH                      | admin/user/{user}                                           | admin::user.admin.user.update           | App\Http\Controllers\Admin\UsersController@update
|        | GET|HEAD                       | admin/user/{user}/edit                                      | admin::user.admin.user.edit             | App\Http\Controllers\Admin\UsersController@edit
```

I propose for the second output to look as follows after assigning an `as` value, matching the functionality of `group` (as resource routes are essentially just a group of routes):

```
|        | GET|HEAD                       | admin/user                                                  | admin::user.index            | App\Http\Controllers\Admin\UsersController@index
|        | POST                           | admin/user                                                  | admin::user.store            | App\Http\Controllers\Admin\UsersController@store
|        | GET|HEAD                       | admin/user/create                                           | admin::user.create           | App\Http\Controllers\Admin\UsersController@create
|        | DELETE                         | admin/user/{user}                                           | admin::user.destroy          | App\Http\Controllers\Admin\UsersController@destroy
|        | GET|HEAD                       | admin/user/{user}                                           | admin::user.show             | App\Http\Controllers\Admin\UsersController@show
|        | PUT|PATCH                      | admin/user/{user}                                           | admin::user.update           | App\Http\Controllers\Admin\UsersController@update
|        | GET|HEAD                       | admin/user/{user}/edit                                      | admin::user.edit             | App\Http\Controllers\Admin\UsersController@edit
```

Currently, you have to use a `names` value which ends up making your routes.php file long and ugly like so:

    Route::group(['prefix' => 'admin', 'as' => 'admin::'], function () {
        Route::resource('user', 'UsersController', [
            'names' => [
                'index' => 'users.index',
                'create' => 'user.create',
                'show' => 'user.show',
                'store' => 'user.store',
                'edit' => 'user.edit',
                'destroy' => 'user.destroy',
                'update' => 'user.update'
            ]
        ]);
    });

Which will give:

```
|        | GET|HEAD                       | admin/user                                                  | admin::users.index                 | App\Http\Controllers\Admin\UsersController@index
|        | POST                           | admin/user                                                  | admin::user.store                  | App\Http\Controllers\Admin\UsersController@store
|        | GET|HEAD                       | admin/user/create                                           | admin::user.create                 | App\Http\Controllers\Admin\UsersController@create
|        | DELETE                         | admin/user/{user}                                           | admin::user.destroy                | App\Http\Controllers\Admin\UsersController@destroy
|        | GET|HEAD                       | admin/user/{user}                                           | admin::user.show                   | App\Http\Controllers\Admin\UsersController@show
|        | PUT|PATCH                      | admin/user/{user}                                           | admin::user.update                 | App\Http\Controllers\Admin\UsersController@update
|        | GET|HEAD                       | admin/user/{user}/edit                                      | admin::user.edit                   | App\Http\Controllers\Admin\UsersController@edit
```

The current functionality could possibly be moved to a separate, more clear, option (`prefix`)

**Comment by narsariamanu** [comment]
@GrahamCampbell Why was this issue closed? It is a duplicate #12597

**Comment by w4** [comment]
@narsariamanu: this issue was closed due to it being fixed in the pull request #12072
